src/ts/components/subtitleoverlay.ts

```diff
@@ -241,104 +244,115 @@ export class SubtitleOverlay extends Container<SubtitleOverlayConfig> {
-    this.backgroundColor.a = alpha;
-    this.updateSubtitleLabelCss();
-    this.setItem('backgroundColor', this.backgroundColor.toCSS());
+    this.subtitleStyleSetting.backgroundColor.a = alpha;
+    this.applyStyleToLabels();
+    this.saveSubtitleStyle();
   }
 
   setWindowOpacity(alpha: number): void {
-    this.windowColor.a = alpha;
-    this.getDomElement().css('background', this.windowColor.toCSS());
-    this.setItem('windowColor', this.windowColor.toCSS());
+    this.subtitleStyleSetting.windowColor.a = alpha;
+    this.getDomElement().css('background', this.subtitleStyleSetting.windowColor.toCSS());
+    this.saveSubtitleStyle();
   }
 
   setFontFamily(fontFamily: string): void {
-    this.fontFamily = fontFamily;
-    this.setItem('fontFamily', this.fontFamily);
-    this.updateSubtitleLabelCss();
+    this.subtitleStyleSetting.fontFamily = fontFamily;
+    this.applyStyleToLabels();
+    this.saveSubtitleStyle();
   }
 
   setFontStyle(fontStyle: string): void {
-    this.fontStyle = fontStyle;
-    this.setItem('fontStyle', this.fontStyle);
-    this.updateSubtitleLabelCss();
+    this.subtitleStyleSetting.fontStyle = fontStyle;
+    this.applyStyleToLabels();
+    this.saveSubtitleStyle();
   }
 
   setFontVariant(fontVariant: string): void {
-    this.fontVariant = fontVariant;
-    this.setItem('fontVariant', this.fontVariant);
-    this.updateSubtitleLabelCss();
+    this.subtitleStyleSetting.fontVariant = fontVariant;
+    this.applyStyleToLabels();
+    this.saveSubtitleStyle();
   }
 
   /**
    * A helper method to avoid updating the CSS label 3 times in a row
    * since family, style and variant are normally updated together
-   **/
+   */
   setFont(fontFamily: string, fontStyle: string, fontVariant: string): void {
-    this.fontFamily = fontFamily;
-    this.setItem('fontFamily', this.fontFamily);
-    this.fontStyle = fontStyle;
-    this.setItem('fontStyle', this.fontStyle);
-    this.fontVariant = fontVariant;
-    this.setItem('fontVariant', this.fontVariant);
-    this.updateSubtitleLabelCss();
+    this.subtitleStyleSetting.fontFamily = fontFamily;
+    this.subtitleStyleSetting.fontStyle = fontStyle;
+    this.subtitleStyleSetting.fontVariant = fontVariant;
+    this.applyStyleToLabels();
+    this.saveSubtitleStyle();
   }
 
   setCharacterEdge(characterEdge: string): void {
-    this.characterEdge = characterEdge;
-    this.updateSubtitleLabelCss();
-    this.setItem('characterEdge', this.characterEdge);
+    this.subtitleStyleSetting.characterEdge = characterEdge;
+    this.applyStyleToLabels();
+    this.saveSubtitleStyle();
   }
 
   setFontSize(coefficient: number): void {
-    this.fontCoefficient = coefficient;
-    this.updateSubtitleLabelCss();
-    this.setItem('fontCoefficient', this.fontCoefficient.toString());
+    this.subtitleStyleSetting.fontCoefficient = coefficient;
+    this.applyStyleToLabels();
+    this.saveSubtitleStyle();
+  }
+
+  private saveSubtitleStyle(): void {
```

Comment:
Using this means that if one settings is changed by the user all other values, whether it's the normal default values or values set in the player config get stored as user preferences and will overwrite default value next time the player is used on the same domain.

It could cause issues if somebody used the player sometimes with special styling and sometimes not.